source/faq.txt

```diff
@@ -148,6 +148,26 @@ When ``MongoClient.close()`` is called by any thread, the driver
 closes all idle sockets and closes all sockets that are in
 use as they are returned to the pool.
 
+Why does the application throw "IllegalStateException: state should be: open"?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You may encounter this exception if you call an operation on a ``MongoClient``
+instance that closed its connections to MongoDB. The connections could be
```

Comment:
Question: Does the client close the connections or does the application close the connections? It could be helpful to clarify the wording here. 

```suggestion
You may encounter this exception if you call an operation on a ``MongoClient``
instance with a closed connection to MongoDB. The connections could be
```

Reply:
I'm not sure I understand the question since they both are responsible for the closing of the connection depending on perspective. An application must call the code to instruct the client to close. The client must close the connection when that method is called. Could you clarify what you meant?

Reply:
I think I meant to ask whether there were times when a client closes a connection on its own, without the application containing/calling the close method. In any case, i think you can keep the wording as is

Reply:
From what I understand, the client, which I'm assuming is MongoClient, can open and close connections based on load and network state (see [How Does Connection Pooling Work in the Java Driver](https://www.mongodb.com/docs/drivers/java/sync/current/faq/#how-does-connection-pooling-work-in-the-java-driver-)), but when you call an operation, whether a direct one such as ListDatabaseNames() or indirectly such as by calling find() on a MongoCollection, the MongoClient queues it up for the next available connection.

However, if you call close() on the instance, or perhaps another process signals for it to close its connections, it transitions into a state that is no longer "open" and does not accept any further operations.